internal/host/static/repository_host_test.go

```diff
@@ -857,6 +934,82 @@ func TestRepository_ListHosts(t *testing.T) {
 	}
 }
 
+func TestRepository_ListHosts_HostSets(t *testing.T) {
+	conn, _ := db.TestSetup(t, "postgres")
+	rw := db.New(conn)
+	wrapper := db.TestWrapper(t)
+	kms := kms.TestKms(t, conn, wrapper)
+	iamRepo := iam.TestRepo(t, conn, wrapper)
+	_, prj := iam.TestScopes(t, iamRepo)
+
+	// testing for full and empty hostset associations
+	catalogs := TestCatalogs(t, conn, prj.PublicId, 3)
+	catalogA, catalogB, catalogC := catalogs[0], catalogs[1], catalogs[2]
+	hostsA := TestHosts(t, conn, catalogA.PublicId, 3)
+	setA := TestSets(t, conn, catalogA.PublicId, 1)[0]
+	TestSetMembers(t, conn, setA.PublicId, hostsA)
+	for _, h := range hostsA {
+		h.SetIds = []string{setA.PublicId}
+	}
+	hostsB := TestHosts(t, conn, catalogB.PublicId, 3)
+
+	// testing for mixed hosts with individual hostsets and empty hostsets
+	hostsC := TestHosts(t, conn, catalogC.PublicId, 5)
+	hostsC0 := TestHosts(t, conn, catalogC.PublicId, 2)
+	setC := TestSets(t, conn, catalogC.PublicId, 5)
+	for i, h := range hostsC {
+		h.SetIds = []string{setC[i].PublicId}
+		TestSetMembers(t, conn, setC[i].PublicId, []*Host{hostsC[i]})
+	}
+	hostsC = append(hostsC, hostsC0...)
+
+	tests := []struct {
+		name      string
+		in        string
+		want      []*Host
+		wantIsErr errors.Code
+	}{
+		{
+			name: "with-hostsets",
+			in:   catalogA.PublicId,
+			want: hostsA,
+		},
+		{
+			name: "empty-hostsets",
+			in:   catalogB.PublicId,
+			want: hostsB,
+		},
+		{
+			name: "mixed-hostsets",
+			in:   catalogC.PublicId,
+			want: hostsC,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			assert, require := assert.New(t), require.New(t)
+			repo, err := NewRepository(rw, rw, kms)
+			assert.NoError(err)
+			require.NotNil(repo)
+			got, err := repo.ListHosts(context.Background(), tt.in)
+			require.NoError(err)
+			// Make sure outputs match. Ignore timestamps.
```

Comment:
Nit: This comment doesn't provide any additional details and simply states what the next line does.  Please remove.

Reply:
Yeah. I copied this comment over from a different test which probably looked messier